src/main/scala/com/fulcrumgenomics/util/Sequences.scala

```diff
@@ -37,6 +37,54 @@ object Sequences {
   /** Common contig/chrom names for non-autosomal sequences in mammals. */
   val CommonNonAutosomalContigNames = Seq("M", "chrM", "MT", "X", "chrX", "Y", "chrY")
 
+  /** Array for determining what bases are compatible. Indices are all DNA/RNA bases. Contents
+    * are Int masks with a bit set for each possible base.  U is treated as identical to T.
+     */
+  private val IupacMasks: Array[Int] = {
+    val masks = new Array[Int](Byte.MaxValue)
+    val (a, c, g, t) = (1, 2, 4, 8)
+
+    def up(c: Char): Int = c.toUpper.toInt
+    def down(c: Char): Int = c.toLower.toInt
+
+    Seq(up(_), down(_)).foreach { f =>
+      masks(f('A')) = a
+      masks(f('C')) = c
+      masks(f('G')) = g
+      masks(f('T')) = t
+      masks(f('U')) = t
+      masks(f('M')) = a | c
+      masks(f('R')) = a | g
+      masks(f('W')) = a | t
+      masks(f('S')) = c | g
+      masks(f('Y')) = c | t
+      masks(f('K')) = g | t
+      masks(f('V')) = a | c | g
+      masks(f('H')) = a | c | t
+      masks(f('D')) = a | g | t
+      masks(f('B')) = c | g | t
+      masks(f('N')) = a | c | g | t
+    }
+
+    masks
+  }
+
+  /** Returns true if two bases are compatible.  Compatibility is defined as having any overlap in the
+    * set of acceptable bases.  E.g.:
+    *   - `compatible('T', 't') == true`
+    *   - `compatible('T', 'U') == true`
+    *   - `compatible('A', 'R') == true`
+    *   - `compatible('M', 'S') == true`
+    *   - `compatible('A', 'C') == false`
+    *   - `compatible('M', 'K') == false`
+    *
+    * @param base1 the first base to be compared
+    * @param base2 the second base to be compared
+    * @return true if the bases share at least one concrete base in common, false otherwise
+    */
+  final def compatible(base1: Byte, base2: Byte): Boolean = (IupacMasks(base1) & IupacMasks(base2)) > 0
```

Comment:
should `compatible('?'.toByte, ?'.toByte)` be allowed (i.e. should it throw an exception?  You could check if IupacMasks(base1/2) returns 0.  Up to you.

Reply:
I waffled on this.  I've changed it to `base1 == base2 || ...` which will allow exact matches for characters we don't expect, and probably also make it more efficient when comparing mostly-identical sequences.